### ISSUE TITLE
Unresolved requirement: gson version="[2.8.9,2.9.0)"

### ISSUE THREAD

**Issue by sebthom** [eclipse/lsp4e#162]
Trying to install LSP4E from https://download.eclipse.org/lsp4e/releases/latest/ results in
```python
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:2.7.3:validate-classpath (default-validate-classpath) on project org.haxe4e: Execution default-validate-classpath of goal org.eclipse.tycho:tycho-compiler-plugin:2.7.3:validate-classpath failed: org.osgi.framework.BundleException: Bundle org.haxe4e cannot be resolved:org.haxe4e [143]
[ERROR]   Unresolved requirement: Require-Bundle: org.eclipse.lsp4e
[ERROR]     -> Bundle-SymbolicName: org.eclipse.lsp4e; bundle-version="0.13.12.202206011407"; singleton:="true"
[ERROR]        org.eclipse.lsp4e [112]
[ERROR]          Unresolved requirement: Require-Bundle: org.eclipse.lsp4j.jsonrpc; bundle-version="[0.14.0,0.15.0)"
[ERROR]            -> Bundle-SymbolicName: org.eclipse.lsp4j.jsonrpc; bundle-version="0.14.0.v20220526-1518"
[ERROR]               org.eclipse.lsp4j.jsonrpc [116]
[ERROR]                 Unresolved requirement: Import-Package: com.google.gson; version="[2.8.9,2.9.0)"
[ERROR]          Unresolved requirement: Require-Bundle: org.eclipse.lsp4j; bundle-version="[0.14.0,0.15.0)"
[ERROR]            -> Bundle-SymbolicName: org.eclipse.lsp4j; bundle-version="0.14.0.v20220526-1518"
[ERROR]               org.eclipse.lsp4j [115]
[ERROR]                 Unresolved requirement: Import-Package: com.google.gson; version="[2.8.9,2.9.0)"
[ERROR]
[ERROR] -> [Help 1]
```

I guess LSP4J 0.14.0 provided via the update site requires gson < 2.9.0 but we only bundle 2.9.0.

**Comment by sebthom** [comment]
Hmm, the dependency was resolved from another update site...

**Comment by martinlippert** [comment]
Actually, I am facing a similar issue, but I am not sure whether this problem can be solved by providing a p2 repo in the build that contains that version of gson. The lsp4e repos have `com.google.gson` included in version `2.8.9`, so the build should be able to find that version to satisfy the dependency requirement from lsp4e (or lsp4j).

The problem seems to occur as soon as I include the staging repo for the Eclipse 2022-09 release, which contains version `2.9.0` of `com.google.gson`. I don't understand yet why this results in the above error, so this is more guesswork than anything else, but it seems to be related.

Any thoughts/ideas?

**Comment by sebthom** [comment]
I get the issue with Eclipse 2022-06 when trying to build a plugin with tycho that depends on LSP4E (which requires gson 2.8.9) and the latest TM4E release which requires (gson >2.9.0). Interestingly manually installing LSP4E and TM4E into an Eclipse IDE works fine.
